src/bootstrap/compile.rs

```diff
@@ -462,7 +462,8 @@ pub fn tool(build: &Build, stage: u32, target: &str, tool: &str) {
 
     let mut cargo = build.cargo(&compiler, Mode::Tool, target, "build");
     let dir = build.src.join("src/tools").join(tool);
-    cargo.arg("--manifest-path").arg(dir.join("Cargo.toml"));
+    cargo.arg("--manifest-path").arg(build.src.join("src/Cargo.toml"));
+    cargo.arg("--package").arg(tool);
```

Comment:
Turns out that Cargo tries to find the workspace toml and use it automatically if it is needed.